### ISSUE TITLE
Fix timer/histogram metrics

### ISSUE THREAD

**Issue by pjenvey** [mozilla-services/autopush-rs#41]
autopush-rs's metrics go through dogstatsd vs the direct datadog HTTP API like autopush-py.

For whatever reason:
```
Note: DogStatsD does NOT implement the following from StatsD:

    Gauge deltas (see this issue)
    Timers as a native metric type (though it does support them via histograms)
```

Even though ```Timers, which exist in StatsD, are a sub-set of histograms in DogStatsD.```

I'm not sure why dogstatsd doesn't handle timers for us (I imagine that's what the python datadog is doing for us under the covers/via the HTTP API), so we'll need to switch to the histogram type ourselves

**Comment by pjenvey** [comment]
#42 switched to histogram yet we continue to not see either of types show up on datadog

**Comment by pjenvey** [comment]
The difference w/ dogstatsd is it allows overriding the datadog API host field via a tag: https://help.datadoghq.com/hc/en-us/articles/218349043-How-to-remove-the-host-tag-when-submitting-metrics-via-dogstatsD

We include a host tag w/ the lifespan metric. In rust the value's the node's internal IP address. It looks like this causes other tags (most importantly 'language:rust' and 'env:prod') to become disassociated with this overridden host (the metrics are there under these overriden raw IPs host)

Python autopush specifies the Host header from the request as this value instead, always something like "push.services.mozilla.com". The python datadog lib does something different here, according to the datadog ui it seems to duplicate the metrics for both the host tag we specify and its regular host value (ec2 instance id).

Long story short: we shouldn't include a host tag in either impl., "push.services.mozilla.com" isn't a useful data point. Nor is it useful in our Session INFO log. The MozLogs format includes a Hostname field anyway (exact value of datadog's)
